src/axom/quest/ScatteredInterpolation.hpp

```diff
@@ -244,6 +247,107 @@ class ScatteredInterpolation
   using PointType = typename DelaunayTriangulation::PointType;
   using BoundingBoxType = typename DelaunayTriangulation::BoundingBox;
 
+private:
+  using MortonIndexType = axom::uint64;
+
+  using VertexSet = typename DelaunayTriangulation::IAMeshType::VertexSet;
+  using VertexIndirectionSet =
+    slam::CoreArrayIndirectionSet<typename VertexSet::PositionType, axom::IndexType>;
+
+private:
+  /**
+   *  \brief Helper struct for sorting input points in the Biased Randomized Incremental Order (BRIO)
+   *
+   *  BRIO helps improve worst-case performance on poorly ordered point sets.
```

Comment:
Skimming the paper, it looks to me that the performance impact depends on the size of the data set. Have you looked at the dependence on performance across a range of data set sizes?

Reply:
Thanks @rhornung67 -- The main benefit is to avoid `O(n^2)` worst case performance across various inputs. In practice, BRIO appears to give `~O(n)` performance. I was hoping this would fix degeneracies I was seeing in a dataset from @dtaller, but unfortunately, it did not, so I'll have to look into an alternate fix.

Regarding performance, I posted some stats on the original Delaunay PR here: https://github.com/LLNL/axom/pull/742
I'm planning to add a similar table to this PR.

Reply:
@rhornung67 and @dtaller -- I posted some performance results in the PR description.

The BRIO yields a (roughly) constant insertion rate when generating Delaunay complexes of different sizes. Previously, the insertion rate was steadily decreasing for increasing input sizes. The code is currently generating 2D Delaunay complexes at a rate of about 270K pts/second, and 3D Delaunay complexes at a rate of ~50K pts/second.

Since the query points are incoherent, this branch does not significantly change the query/interpolation rate.